optuna/_gp/batched_lbfgsb.py

```diff
@@ -102,12 +103,38 @@ def batched_lbfgsb(
     max_iters: int = 15000,
     max_line_search: int = 20,
 ) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
-
+    """
+    Batched L-BFGS-B optimization with/without greenlet.
+    - `func_and_grad` is expected to take a 2D array as the first argument and return a tuple of
+      a 1D array of function values and a 2D array of gradients.
+    - `x0_batched` is a 2D array where each row is an initial point for optimization.
+    - `args_tuple` is a tuple of additional arguments to pass to `func_and_grad`. e.g., if
+      `args_tuple` is `([alpha1, ..., alphaB], [beta1, ..., betaB])`, then
+      `func_and_grad` is called as
+      `func_and_grad(x0_batched, [alpha1, ..., alphaB], [beta1, ..., betaB])`. Note that each
+      argument in `args_tuple` is expected to be a list of length `B` (batch size).
+    """
+    assert x0_batched.ndim == 2
+
+    batch_size, dim = x0_batched.shape
+    # Validate args_tuple shapes: each arg must be a sequence of length B.
+    for j, arg in enumerate(args_tuple):
+        if len(arg) != batch_size:
+            raise ValueError(f"args_tuple[{j}] must have length {batch_size}, got {len(arg)}.")
```

Comment:
`batch_size, dim = x0_batched.shape` checks whether `x0_batched` is 2D.

`X must be .... Got Y` or `X must be ..., but got Y.` is more common.
Optuna mostly uses the latter.

```suggestion
    batch_size, dim = x0_batched.shape
    # Validate args_tuple shapes: each arg must be a sequence of length B.
    for j, arg in enumerate(args_tuple):
        if len(arg) != batch_size:
            raise ValueError(f"args_tuple[{j}] must have length {batch_size}, but got {len(arg)}.")
```

Reply:
I understand. Thank you!